Source/WebCore/css/CSSCursorImageValue.h

```diff
@@ -35,12 +36,10 @@ class BuilderState;
 
 class CSSCursorImageValue final : public CSSValue {
 public:
-    static Ref<CSSCursorImageValue> create(Ref<CSSValue>&& imageValue, const std::optional<IntPoint>& hotSpot, LoadedFromOpaqueSource);
-    static Ref<CSSCursorImageValue> create(Ref<CSSValue>&& imageValue, const std::optional<IntPoint>& hotSpot, URL, LoadedFromOpaqueSource);
+    static Ref<CSSCursorImageValue> create(Ref<CSSValue>&& imageValue, RefPtr<CSSValue>&& hotSpot, LoadedFromOpaqueSource);
+    static Ref<CSSCursorImageValue> create(Ref<CSSValue>&& imageValue, RefPtr<CSSValue>&& hotSpot, URL, LoadedFromOpaqueSource);
```

Comment:
I wish we had a way to express that this hot spot value is known to be a non-coalescing pair of numbers: In the type system or even with comments and assertions. For example, should this at least be `CSSValuePair` instead of just `CSSValue`?

What I don’t like is that the parser has the checking responsibility, but this code has the resolution responsibility that relies on that checking. And the computed style code has "checking" responsibility for making sure it creates the same type of object that also has to work with the resolution code. I would like to make it likely that future programmers won’t accidentally use a type other than a non-coalescing pair of `CSSPrimitiveValue` numbers, and will immediately understand they made a mistake even if there isn’t thorough test coverage.